File: hackerank/Easy/counting_valleys/solutions/python/solution.test.py

from solution import counting_valleys


def test_counting_valleys():
    path1, steps1, paths2, step2 = "DDUUDDUDUUUD", 12, "UDDDUDUU", 8
    valleys1, valleys2 = counting_valleys(
        steps1, path1), counting_valleys(step2, paths2)
    assert(valleys1) == 2, "Expected the number of valleys to be 2"
    assert(valleys2) == 1, "Expected the number of valleys to be 1"


if __name__ == "__main__":
    test_counting_valleys()
    print("PASS \n ok")

File: hackerank/Easy/counting_valleys/solutions/go/solution._test.go
package countingvalleys

import "testing"

func TestCountingValleys(t *testing.T) {
	path1, steps1, paths2, step2 := "DDUUDDUDUUUD", 12, "UDDDUDUU", 8
	valleys1, valleys2 := countingValleys(steps1, path1), countingValleys(step2, paths2)

	if valleys1 != 2 {
		t.Errorf("Expected 2 but received %v", valleys1)
	}

	if valleys2 != 1 {
		t.Errorf("Expected 1 but received %v", valleys2)
	}
}
